### DISCUSSION TITLE
R - Data -Inf pada dataframe

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#56] [Q&A] [answered]
Apabila terdapat data -Inf pada sebuah dataframe akibat logtransformation pada numerik bernilai 0, 
apakah sebaiknya -Inf tersebut kemudian diimputasi dengan nilai 0?

**Comment by tomytjandra** [comment, marked as answer]
Halo Pak @Hertzspr, secara matematis nilai logaritma dari 0 tidak terdefinisi, pada R nilai `log(0) = -Inf`. Terdapat beberapa pendekatan yang bisa kita lakukan apabila dataset kita memiliki nilai 0 lalu kita ingin terapkan fungsi transformasi:

1. Gunakan fungsi transformasi lain yang terdefinisi untuk nilai 0, misalnya fungsi akar kuadrat `sqrt()` karena akar dari 0 adalah 0 (terdefinisi)
2. Gunakan fungsi `log1p()` yang akan mengkomputasikan nilai log(1 + x), sehingga `log1p(0) = 0`

Semoga bermanfaat 😄

**Reply by Hertzspr** [reply]
Oiya mas kalo sqrt() kan akar pangkat 2 ya, kalo mau buat akar pangkat 3 gimana?

**Reply by tomytjandra** [reply]
Untuk akar pangkat n, kita dapat menggunakan fungsi `nthroot()` dari library `pracma`. Contohnya sebagai berikut:

```
library(pracma)
number <- c(0, -1, 8, -27)
nthroot(number, 3)
```

**Output:** `0 -1  2 -3`
